### PR TITLE
Added another variable of total cows bought with functionality

### LINKED ISSUE #4: FEATURE: Track total cows bought
As a player/admin I can see the total cows bought for just me (or as an admin, for each user)
So that I can better understand what's happening in the game

### Discussion
In this issue we add to the user commons model one new field
- total cows bought

Then we update the logic that buys cows so that the field is updated appropriately.  Cows bought is cumulative, so we only add to this field, never subtract.

Then we want to add these fields to the UI for the individual player, and to the leaderboard.

### Acceptance Criteria
- [ ] cows bought updates appropriately when cows are bought.
- [ ] field is visible on the play page for the player
- [ ] field is visible on the leaderboard

### Implementation Todos
- Add field to userCommons model
- Add logic to update field for buy backend
- Add frontend display logic to play page
- Add frontend display logic to the leaderboard table

### PR DESCRIPTION
**Pull request by jgarcia-ucsb**
Closes [#4](https://github.com/orgs/ucsb-cs156-s23/projects/64?pane=issue&itemId=28963915) Added a cows bought variable that tracks the total amount of cows a user buys over the time the game is played. Helps the user track the amount of cows they ended up buying by the end of the game. (Note: the jpegs did not load in localhost, but I tested dokku qa and it worked there)

Before (Leaderboard)
BeforeLeaderboardHC

After(Leaderboard)
AfterLeaderBoardHC

Before(Play Page)
BeforePageHC

After(Play Page)
AfterPageHC

Before(Database)
BeforeDBHC

After(Database)
AfterDBHC

### CODE REVIEW

File: src/main/java/edu/ucsb/cs156/happiercows/entities/UserCommons.java [hunk 1]
```diff
@@ -33,5 +33,7 @@ public class UserCommons {
   private int numOfCows;
 
   private double cowHealth;
+
+  private int cowsBought;
```
Comment (chriscerie) [inline src/main/java/edu/ucsb/cs156/happiercows/entities/UserCommons.java]: It's not immediately obvious for new readers what the difference between `numOfCows` and `cowsBought` are as they can both mean either things. Something like `lifetimeNumCows` would be more readable.

**Review by chriscerie — CHANGES_REQUESTED** [review]
Thanks!

Validated stories:

https://ucsb-cs156-s23.github.io/proj-happycows-s23-6pm-2/prs/17/storybook/?path=/story/components-leaderboard-leaderboardtable--five-user-commons

https://ucsb-cs156-s23.github.io/proj-happycows-s23-6pm-2/prs/17/storybook/?path=/story/components-commons-managecows--uncontrolled

**Comment by jgarcia-ucsb** [comment]
Changed all instances of 'cowsBought' to 'lifetimeNumCows' in order to make the code more readable and accessible for anybody inheriting the code. The functionality stays the same as the previous pr version as that wasn't the problem.

The database table now looks like this:
Screen Shot 2023-06-01 at 9 07 25 AM

**Review by chriscerie — APPROVED** [review]

**Review by pconrad — APPROVED** [review]
LGTM

File: src/main/java/edu/ucsb/cs156/happiercows/entities/UserCommons.java (hunk 1 above)
Comment (pconrad) [reply src/main/java/edu/ucsb/cs156/happiercows/entities/UserCommons.java]: Chris has a good point.   I think perhaps you should change this.

**Review by pconrad — CHANGES_REQUESTED** [review]
Please make the changes that Chris requested.  This is in pretty good shape otherwise.

**Pull request closed without merging**